plenum/server/batch_handlers/node_reg_handler.py

```diff
@@ -79,7 +70,10 @@ def post_batch_rejected(self, ledger_id, prev_handler_result=None):
     def commit_batch(self, three_pc_batch: ThreePcBatch, prev_handler_result=None):
         prev_committed = self.committed_node_reg
         self.committed_node_reg = self._uncommitted.popleft().uncommitted_node_reg
-        self._committed_view_no = three_pc_batch.view_no if three_pc_batch.original_view_no is None else three_pc_batch.original_view_no
+        three_pc_batch_view_no = three_pc_batch.view_no if three_pc_batch.original_view_no is None else three_pc_batch.original_view_no
+        if three_pc_batch_view_no > self._committed_view_no:
+            self.node_reg_at_beginning_of_view[three_pc_batch_view_no] = list(self.committed_node_reg)
+        self._committed_view_no = three_pc_batch_view_no
```

Comment:
May this line should be under `if` condition? If now is reordering phase after view_change then `self.node_reg_at_beginning_of_view[three_pc_batch_view_no]` will be rewritten on each batch and `self._committed_view_no` will be dropped to previous view_no.

Reply:
Yes, I agree that it will be more clear to move it under `if` condition. 
However, even the current code should not lead to any issues, since `commit_batch` is called from non-committed (previously non-ordered) batches only. So, if this is re-ordering after view change phase, this will be called only if a batch is not ordered in the previous view. If so, the current `_committed_view_no` can not be higher yet as we don't allow gaps in PrePrepares (we can not Commit in the new view until we finish re-ordering of all batches from the previous view).